include/gridtools/sid/simple_ptr_holder.hpp

```diff
@@ -29,15 +30,10 @@ namespace gridtools {
         GT_TARGET_NAMESPACE {
             template <class T>
             struct simple_ptr_holder {
-                T m_val;
+                ldg_ptr_t<T> m_val;
```

Comment:
I am confused. Why is this change safe? In general we can write to a ptr hold by a simple_ptr_holder?
What I had in mind was just a simple function

Reply:
Did you check the code for `ldg_ptr_t`? `ldg_ptr_t<T>` is just `T` for all types except for those which support LDG (that is `float const*` etc.), then it will be a wrapper type (`impl_::ldg_ptr<T>`). So I guess it is the return type of this simple function you had in mind?

Reply:
I didn't parse it correctly, but if I would have read the tests, I could have answered my question...

Reply:
I moved the application of `as_ldg_ptr` into `operator()` to keep the CTAD logic, so `ldg_ptr_t` is gone anyway now.